sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_do_while.py

```diff
@@ -38,8 +38,8 @@ def pipeline_with_do_while_node():
 
     :param body: Pipeline job for the do-while loop body.
     :type body: Pipeline
-    :param condition: Name of boolean output of body as do-while loop condition.
-    :type condition: Output
+    :param condition: Either name of boolean output of body or simply True as do-while loop condition.
```

Comment:
what if user pass in a False here?
Ideally we only accept True?

Reply:
I think only True is better, will update here and schema to ban False from statement/load (currently this happens during validation)

Reply:
I have updated one version with following validation logic:

- SDK: there is a check in `do_while` to forbid while `False`, otherwise an user error will be raised.
- YAML: validate on the level of schema to forbid load negative value to `condition` field; however, `condition` is an `UnionField` and due to marshmallow mechanism, the error message will look like below:

![image](https://user-images.githubusercontent.com/38847871/215419566-7f9e0b4d-85ab-4455-8a8f-807f74b373fb.png)

Not sure if it is acceptable as the error message is kind of redundant.

Reply:
Have wrapped a schema class `DoWhileConditionSchema` for condition field to raise error with better error message.

Reply:
Reverted related changes to avoid confusing usage.